### PR TITLE
feat(menu-trigger): bug fixes

### PR DESCRIPTION
**Pull request by torgeadelin**
# Description
- the internal state of the menuTrigger doesn't reflect correctly when popover was closed via click outside
- warnings about type of children in MenuTrigger, removed
- children type for MenuTrigger is `ReactElement[]` which is wrong, it's supposed to be `ReactNode`. 
- updated docs for MenuTrigger

### CODE REVIEW

File: src/components/MenuTrigger/MenuTrigger.types.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { CSSProperties, ReactElement } from 'react';
+import { CSSProperties, ReactElement, ReactNode } from 'react';
```
Comment (InteractiveTimmy) [inline src/components/MenuTrigger/MenuTrigger.types.ts:1]: Aren't these all types? Could we use `type` keyword?

File: src/components/Popover/Popover.tsx [hunk 2]
```diff
@@ -70,6 +71,7 @@ const Popover: FC<Props> = (props: Props) => {
       trigger={trigger === 'mouseenter' ? 'mouseenter focusin' : trigger}
       interactive={interactive}
       appendTo="parent"
+      onClickOutside={onClickOutside}
```
Comment (InteractiveTimmy) [inline src/components/Popover/Popover.tsx:74]: Strange workaround for popover, i thought tippy supported this already?

**Review by InteractiveTimmy — CHANGES_REQUESTED** [review]
Just a small item around type imports, otherwise, looks good!

File: src/components/Popover/Popover.tsx (hunk 2 above)
Comment (torgeadelin) [reply src/components/Popover/Popover.tsx:74]: It's not a workaround, this is from tippy, we just didn't use it.

**Review by InteractiveTimmy — APPROVED** [review]
LGTM!

**Pull request merged**